### ISSUE TITLE
submitting FieldArray

### ISSUE THREAD

**Issue by pdavidow** [erikras/redux-form#2275]
I am grappling trying to understand what should be done upon submit for a FieldArray.  Offhand, I could simply update the store with the **entire** array that the form provides. But the whole idea of Redux is to document **incremental** changes.  If I have an array of 100 items, and the user updates/deletes just one, that's what I'd expect the store to archive -- rather than setting the entire array anew.  

I realize that ReduxForm itself actually interacts with the store incrementally, but that is for display purposes only.  Could/should that information  be transferred to the 'real' (i.e., non-UI related) store reducer? Or perhaps setting the entire array is fine, being that the ReduxForm interactivity itself is archived. 

In summary: What should be done upon submit for a FieldArray?

**Comment by gustavohenke** [comment]
Is your back-end an event-source based system (perhaps Redux itself)?
If so, you should probably listen to the Redux Form actions and submit these instead of the consolidated values:

```
POST /api/resource/1/actions

[
  { "type": "remove", "meta": { "path": "values[1]" } },
  { "type": "change", "meta": { "path": "values[0].foo" }, "payload": "bar" },
  { "type": "add", "meta": {  "path": "values" } }
]
```

Otherwise, I see no problem in storing the values at once.

**Comment by gustavohenke** [comment]
ping

**Comment by pdavidow** [comment]
no back-end at this point...

**Comment by gustavohenke** [comment]
Okay, so you could anyway deal with your own reducers incrementally as I told you above.

To perhaps give you yet another idea:
The backend for one of my apps deals with form data in an event-sourced way, just like Redux: it receives all data, everytime, but we only store the difference between each save.
This way we have the "time traveling" feature that Redux made possible for JS apps 🎉
